client/src/components/UsedOnTable/provideUsedOnData.js

```diff
@@ -1,51 +1,47 @@
-import React, { Component } from 'react';
+import React, { useEffect, useState } from 'react';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import { loadUsedOn } from 'state/usedOn/usedOnActions';
 import { injectTabContext } from 'hooks/useTabContext';
 
 const provideUsedOnData = (UsedOnTable) => {
-  class UsedOnDataProvider extends Component {
-    componentDidMount() {
-      const firstMount = this.haveFetchedData === undefined;
-      const tabContext = this.props.tabContext ? this.props.tabContext : false;
-      const activeTab = tabContext ? tabContext.isOnActiveTab : false;
-      this.haveFetchedData = false;
-      if ((firstMount && activeTab) || this.props.forceFetch) {
-        this.fetchDataFromEndpoint();
-      }
-    }
+  const UsedOnDataProvider = ({
+    identifier,
+    loading,
+    data,
+    error,
+    usedOn,
+    forceFetch,
+    tabContext,
+    loadUsedOn: loadUsedOnAction,
+  }) => {
+    const [haveFetchedData, setHaveFetchedData] = useState(false);
 
-    componentDidUpdate(oldProps) {
-      const tabContext = this.props.tabContext;
+    const fetchDataFromEndpoint = () => {
+      const { method, url } = (data.readUsageEndpoint || {});
+      if (!haveFetchedData || forceFetch) {
+        // see client/src/state/usedOn/usedOnActions.js
+        loadUsedOnAction(identifier, method, url);
+      }
+      setHaveFetchedData(true);
+    };
 
-      // Fetch data if
+    useEffect(() => {
+      // Fetch data if on of the following is true:
       // - identifier has changed
       // - component is not in a tab
       // - component is on the currently active tab
-      const doFetch =
-        oldProps.identifier !== this.props.identifier ||
-        !tabContext ||
-        tabContext.isOnActiveTab;
-
-      if (doFetch) {
-        this.fetchDataFromEndpoint();
+      if (!tabContext || (tabContext && tabContext.isOnActiveTab)) {
```

Comment:
```suggestion
      if (!tabContext || tabContext.isOnActiveTab) {
```
`tabContext` must be truthy for `!tabContext` to evaluate false, so we don't need to check it a second time - won't hold up merging just for this though.